# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserParent.h
# Commit: 405100db6c45
# Full Hash: 405100db6c45f0463286067df5b491f74790d9aa
# Author: Kashav Madan <kmadan@mozilla.com>
# Date: 2019-07-11 09:51:12
# Regressor Bug: 1543251
# File Overlap Count: 1
# Description:
#   Bug 1543251 - Move hasBeforeUnload from PBrowser to PWindowGlobal, r=nika
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D37003
# ==============================================================================

diff -r 0c408343080d -r 405100db6c45 dom/ipc/BrowserParent.h
--- a/dom/ipc/BrowserParent.h	Thu Jul 04 07:41:57 2019 +0000
+++ b/dom/ipc/BrowserParent.h	Wed Jul 10 21:13:44 2019 +0000
@@ -274,8 +274,6 @@
   mozilla::ipc::IPCResult RecvAccessKeyNotHandled(
       const WidgetKeyboardEvent& aEvent);
 
-  mozilla::ipc::IPCResult RecvSetHasBeforeUnload(const bool& aHasBeforeUnload);
-
   mozilla::ipc::IPCResult RecvRegisterProtocolHandler(const nsString& aScheme,
                                                       nsIURI* aHandlerURI,
                                                       const nsString& aTitle,
@@ -716,7 +714,6 @@
   void Deprioritize();
 
   bool GetHasContentOpener();
-  bool GetHasBeforeUnload();
 
   bool StartApzAutoscroll(float aAnchorX, float aAnchorY, nsViewID aScrollId,
                           uint32_t aPresShellId);
@@ -942,10 +939,6 @@
   // at least once.
   bool mHasPresented;
 
-  // True if at least one window hosted in the BrowserChild has added a
-  // beforeunload event listener.
-  bool mHasBeforeUnload;
-
   // True when the remote browser is created and ready to handle input events.
   bool mIsReadyToHandleInputEvents;
 